src/query/api/v1/handler/prometheus/native/read_instantaneous.go

```diff
@@ -96,7 +96,9 @@ func (h *PromReadInstantHandler) ServeHTTP(w http.ResponseWriter, r *http.Reques
 		QueryContextOptions: models.QueryContextOptions{
 			LimitMaxTimeseries: fetchOpts.Limit,
 		}}
-	if restrictOpts := fetchOpts.RestrictFetchOptions; restrictOpts != nil {
+
```

Comment:
Does the result from `read(...)` not need to be filtered out too with `prometheus.FilterSeriesByOptions(result.series, fetchOpts)`?

Reply:
Moved it to a common file.